pkg/engine2/reconciler/reconciler.go

```diff
@@ -3,65 +3,195 @@ package reconciler
 	construct "github.com/klothoplatform/klotho/pkg/construct2"
 	"github.com/klothoplatform/klotho/pkg/engine2/solution_context"
 	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
 	"github.com/klothoplatform/klotho/pkg/set"
 	"go.uber.org/zap"
 )
 
-func RemoveResource(c solution_context.SolutionContext, resource construct.ResourceId, explicit bool) error {
-	zap.S().Debugf("reconciling removal of resource %s ", resource)
-	upstreams, downstreams, err := construct.Neighbors(c.DataflowGraph(), resource)
-	if err != nil {
-		return err
+type (
+	deleteRequest struct {
+		resource construct.ResourceId
+		explicit bool
 	}
+)
 
-	template, err := c.KnowledgeBase().GetResourceTemplate(resource)
-	if err != nil {
-		return fmt.Errorf("unable to remove resource: error getting resource template for %s: %v", resource, err)
-	}
-	canDelete, err := canDeleteResource(c, resource, explicit, template, upstreams, downstreams)
-	if err != nil {
-		return err
-	}
-	if !canDelete {
-		return nil
-	}
+func RemoveResource(c solution_context.SolutionContext, resource construct.ResourceId, explicit bool) error {
+	zap.S().Debugf("reconciling removal of resource %s ", resource)
 
-	op := c.OperationalView()
+	queue := make([]deleteRequest, 0)
+	queue = append(queue, deleteRequest{
+		resource: resource,
+		explicit: explicit,
+	})
 
 	var errs error
-	// We must remove all edges before removing the vertex
-	for res := range upstreams {
-		errs = errors.Join(errs, op.RemoveEdge(res, resource))
-	}
-	for res := range downstreams {
-		errs = errors.Join(errs, op.RemoveEdge(resource, res))
-	}
-	if errs != nil {
-		return errs
+
+	for len(queue) > 0 {
+		request := queue[0]
+		queue = queue[1:]
+		resource := request.resource
+		explicit := request.explicit
+
+		upstreams, downstreams, err := construct.Neighbors(c.DataflowGraph(), resource)
+		if err != nil {
+			errs = errors.Join(errs, err)
+			continue
+		}
+
+		template, err := c.KnowledgeBase().GetResourceTemplate(resource)
+		if err != nil {
+			errs = errors.Join(errs, fmt.Errorf("unable to remove resource: error getting resource template for %s: %v", resource, err))
+			continue
+		}
+		canDelete, err := canDeleteResource(c, resource, explicit, template, upstreams, downstreams)
+		if err != nil {
+			errs = errors.Join(errs, err)
+			continue
+		}
+		if !canDelete {
+			continue
+		}
+		namespacedResources, err := findAllResourcesInNamespace(c, resource)
+		if err != nil {
+			errs = errors.Join(errs, err)
+			continue
+		}
+
+		op := c.OperationalView()
+
+		// We must remove all edges before removing the vertex
+		for res := range upstreams {
+			errs = errors.Join(errs, op.RemoveEdge(res, resource))
+		}
+		for res := range downstreams {
+			errs = errors.Join(errs, op.RemoveEdge(resource, res))
+		}
+
+		err = construct.RemoveResource(op, resource)
+		if err != nil {
+			errs = errors.Join(errs, err)
+			continue
+		}
+
+		for _, res := range namespacedResources.ToSlice() {
```

Comment:
```suggestion
		for res := range namespacedResources {
```
nit: don't need to copy the set to iterate it